drivers/ipm/ipm_rpi_pico.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2025 Dan Collins
+ * Copyright (c) 2025 Dmitrii Sharshakov <d3dx12.xx@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT raspberrypi_pico_sio_fifo
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/ipm.h>
+
+#include <zephyr/logging/log.h>
+#include "hardware/structs/sio.h"
+
+LOG_MODULE_REGISTER(ipm_rpi_pico, CONFIG_IPM_LOG_LEVEL);
+
+struct rpi_pico_mailbox_config {
+	sio_hw_t *const sio_regs;
+};
+
+struct rpi_pico_ipm_data {
+	ipm_callback_t cb;
+	void *user_data;
+};
+
+static struct rpi_pico_mailbox_config rpi_pico_mailbox_config = {
+	.sio_regs = (sio_hw_t *)DT_INST_REG_ADDR_BY_NAME(0, sio),
+};
+static struct rpi_pico_ipm_data rpi_pico_mailbox_data;
+
+static int rpi_pico_mailbox_send(const struct device *dev, int wait, uint32_t id, const void *data,
```

Comment:
I had to look at the docs for ipm_send() to figure out that `int wait` is being used as a boolean. Strange!